### PR TITLE
refactor: move V2 logwriter into internal module

### PR DESCRIPTION
**Pull request by Yun-Kim**
This PR moves the `V2LogWriter` class into `ddtrace.internal.log_writer` from `ddtrace.contrib.openai._logging` as this will be reused by the LangChain integration, and will no longer be specific to the OpenAI integration.

The test file `tests.contrib.openai.test_logging` will remain in the `contrib.openai` module as this reflects the openai integration.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](../docs/contributing.rst#release-branch-maintenance))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](../docs/contributing.rst#release-branch-maintenance)

### CODE REVIEW

File: ddtrace/internal/log_writer.py [hunk 1]
```diff
@@ -72,7 +72,6 @@ def enqueue(self, log):
             self._buffer.append(log)
 
     def on_shutdown(self):
-        # type: (...) -> None
```
Comment (mabdinur) [inline ddtrace/internal/log_writer.py:75]: Should we revert this change?

**Review by mabdinur — APPROVED** [review]

File: ddtrace/internal/log_writer.py (hunk 1 above)
Comment (Yun-Kim) [reply ddtrace/internal/log_writer.py:75]: This is a slightly bigger issue than type hinting - mypy (which isn't enabled for contrib files I believe, so it's only flagging it now that we've moved this to `internal`) flags this as the superclass method is actually a static method (so no reference to `self`) and thus has a different function signature `# type: () -> Any`. 

I think we should probably address this through a redesign, but that would be a lot of work and unnecessary for the scope of this PR. I made the decision to ignore this for now and address it as a future PR.

**Review by mabdinur — APPROVED** [review]
